src/tests/ftest/container/container_redundancy.py

```diff
@@ -0,0 +1,152 @@
+
+    def execute_during_rebuild(self):
+        """Execute test steps during rebuild."""
+        self.daos_cmd = DaosCommand(self.bin)
+        rf = ''.join(self.container.properties.value.split(":"))
+        # Exclude the second rank from the pool during rebuild
+        self.server_managers[0].stop_ranks(
+            [self.inputs.rank.value[1]], self.d_log)
+
+        self.log.info(
+            "==>(3)Check for container rf and health-status "
+            "after the 2nd rank rebuild: HEALTHY")
+        self.verify_cont_rf_healthstatus(rf, "HEALTHY")
+        self.server_managers[0].stop_ranks(
+            [self.inputs.rank.value[2]], self.d_log)
+
+        self.log.info(
+            "==>(4)Check for container rf and health-status "
+            "after the 3rd rank rebuild: UNCLEAN")
+        self.verify_cont_rf_healthstatus(rf, "UNCLEAN")
+
+        # Wait for rebuild completion and set container property healthy
+        self.pool.wait_for_rebuild(False, 1)
+        self.daos_cmd.container_set_prop(
+                      pool=self.pool.uuid,
+                      cont=self.container.uuid,
+                      prop="status",
+                      value="healthy")
+        self.log.info(
+            "==>(5)Check for container rf and health-status "
+            "after rebuild completed: HEALTHY")
+        self.verify_cont_rf_healthstatus(rf, "HEALTHY")
+
+        # Populate the container with additional data after rebuild
+        self.container.write_objects(obj_class=self.inputs.object_class.value)
+
+    def test_container_redundancy_factor_with_rebuild(self):
+        """Jira ID:
+        DAOS-6270: container with RF 2 can lose up to 2 concurrent
+                   servers without err.
+        DAOS-6271: container with RF 2 error reported after more than
+                   2 concurrent servers failure occurred
+        Description:
+            Test step:
+                (0)Create pool and container with redundant factor
+                   Start background IO object write, and server-rank
+                   system stop sequentially.
+                (1)Check for container initial rf and health-status.
+                (2)Check for container rf and health-status after the
+                   1st rank rebuild.
+                (3)Check for container rf and health-status after the
+                   2nd rank rebuild.
+                (4)Check for container rf and health-status after the
+                   3rd rank rebuild.
+                (5)Check for container rf and health-status after the
+                   rebuild completed.
+                (6)Verify container io object write.
+        Use Cases:
+            Verify container RF with rebuild with multiple server failures.
+
+        :avocado: tags=all,full_regression
+        :avocado: tags=container,rebuild
+	:avocado: tags=container_redundancy
+        """
+        self.mode = "container_rf"
+        self.execute_rebuild_test()
```

Comment:
Looks like this should be  execute_during_rebuild... I don't see execute_rebuild_test method,.

Reply:
def execute_rebuild_test method is imported from rebuild_test_base.py.

Reply:
Oh, you are inheriting the methods and overriding them... Good... I am fine...